### PR TITLE
Add support for BGP Hold and Keep Alive timers

### PR DESCRIPTION
**Pull request by SMBurrows**
Added support for the Keep Alive and Hold timers for a BGP Peer.

https://pubhub.devnetcloud.com/media/dme-docs-10-2-2/docs/Routing%20and%20Forwarding/bgp:Peer/

This will allow for 
```
router bgp xxx
  router-id x.x.x.x
  neighbor x.x.x.x
    timers 5 15
```

### CODE REVIEW

File: gen/definitions/bgp_peer.yaml [hunk 1]
```diff
@@ -63,6 +64,24 @@ attributes:
     description: 'Source Interface. Must match first field in the output of `show intf brief`.'
     default_value: 'unspecified'
     example: 'lo0'
+  - nxos_name: holdIntvl
+    tf_name: hold_time
+    type: Int64
+    description: 'BGP Hold Timer in seconds. The value ust be greater than the keep alive timer'
+    default_value: 180
+    example: 5
+    omit_empty_value: true
```
Comment (danischm) [inline gen/definitions/bgp_peer.yaml:73]: What is the reason for using `omit_empty_value` here?

File: gen/definitions/bgp_peer.yaml [hunk 2]
```diff
@@ -63,6 +64,24 @@ attributes:
     description: 'Source Interface. Must match first field in the output of `show intf brief`.'
     default_value: 'unspecified'
     example: 'lo0'
+  - nxos_name: holdIntvl
+    tf_name: hold_time
+    type: Int64
+    description: 'BGP Hold Timer in seconds. The value ust be greater than the keep alive timer'
```
Comment (danischm) [inline gen/definitions/bgp_peer.yaml:70]: typo, `ust` -> `must`

File: gen/definitions/bgp_peer.yaml [hunk 3]
```diff
@@ -63,6 +64,24 @@ attributes:
...
+    default_value: 180
+    example: 5
+    omit_empty_value: true
+    min_int: 3
+    max_int: 3600
+  - nxos_name: kaIntvl
+    tf_name: keep_alive
```
Comment (danischm) [inline gen/definitions/bgp_peer.yaml:77]: Could we maybe change this to `keepalive`?

**Review by danischm — CHANGES_REQUESTED** [review]

**Comment by danischm** [comment]
Many thanks for your contribution!

File: gen/definitions/bgp_peer.yaml (hunk 1 above)
Comment (SMBurrows) [reply gen/definitions/bgp_peer.yaml:73]: I left that in there when I was testing. I was trying to see if that would remove the entry from the device config when it was removed from the HCL. 

I've noticed similar behavior to interface descriptions - if you add a description and deploy, and then remove it from the hcl and deploy the description will remain on the device. 

Do you know how to fix that behavior?

File: gen/definitions/bgp_peer.yaml (hunk 3 above)
Comment (SMBurrows) [reply gen/definitions/bgp_peer.yaml:77]: Sure can!

File: gen/definitions/bgp_peer.yaml (hunk 1 above)
Comment (danischm) [reply gen/definitions/bgp_peer.yaml:73]: Looking at the code it should push an empty description and therefore remove the description. If that is not the case, I would need to investigate further.

File: gen/definitions/bgp_peer.yaml [hunk 4]
```diff
@@ -42,6 +42,7 @@ attributes:
     type: String
     description: 'Peer description.'
     example: 'My description'
+    omit_empty_value: true
```
Comment (danischm) [inline gen/definitions/bgp_peer.yaml:45]: Let's remove it here as well.

File: examples/resources/nxos_bgp_peer/resource.tf [hunk 5]
```diff
@@ -7,4 +7,6 @@ resource "nxos_bgp_peer" "example" {
   peer_template    = "SPINE-PEERS"
   peer_type        = "fabric-internal"
   source_interface = "lo0"
+  hold_time        = 5
+  keep_alive       = 15
```
Comment (danischm) [inline examples/resources/nxos_bgp_peer/resource.tf:11]: I think you need to run `go generate` again to update the code and documentation accordingly.

**Pull request merged**